### PR TITLE
fix: Templates should accept either query or input table

### PR DESCRIPTION
**Pull request by poojabasker20**

### CODE REVIEW

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/google/cloud/dataproc/templates/databases/CassandraToGCSConfig.java [hunk 1]
```diff
@@ -68,6 +68,16 @@ public class CassandraToGCSConfig {
...
   private String sparkLogLevel;
 
+  @AssertTrue(
+      message =
+          "Provide either input query or both input keyspace and table. Refer to databases/README.md for more instructions.")
+  private boolean isInputValid() {
+    return ((StringUtils.isBlank(query)
+            || (StringUtils.isBlank(keyspace) && StringUtils.isBlank(inputTable)))
+        && (StringUtils.isNotBlank(query)
+            || (StringUtils.isNotBlank(keyspace) && StringUtils.isNotBlank(inputTable))));
```
[the remark this answers is not in the document]
Comment (poojabasker20) [reply java/src/main/java/com/google/cloud/dataproc/templates/databases/CassandraToGCSConfig.java]: Updated.

File: java/src/main/java/com/google/cloud/dataproc/templates/snowflake/SnowflakeToGCSConfig.java [hunk 2]
```diff
@@ -144,10 +142,18 @@ public String getSparkLogLevel() {
...
-    return (StringUtils.isBlank(sfTable) || StringUtils.isBlank(sfQuery))
-        && (StringUtils.isNotBlank(sfTable) || StringUtils.isNotBlank(sfQuery));
+    return ((StringUtils.isBlank(sfQuery)
+            || (StringUtils.isBlank(sfDatabase)
+                && StringUtils.isBlank(sfSchema)
+                && StringUtils.isBlank(sfTable)))
+        && (StringUtils.isNotBlank(sfQuery)
+            || (StringUtils.isNotBlank(sfDatabase)
+                && StringUtils.isNotBlank(sfSchema)
+                && StringUtils.isNotBlank(sfTable))));
```
[the remark this answers is not in the document]
Comment (poojabasker20) [reply java/src/main/java/com/google/cloud/dataproc/templates/snowflake/SnowflakeToGCSConfig.java]: Improved readability by breaking the check in two as done in the python template.

File: java/src/main/java/com/google/cloud/dataproc/templates/snowflake/SnowflakeToGCSConfig.java [hunk 3]
```diff
@@ -144,10 +142,26 @@ public String getSparkLogLevel() {
...
+  @AssertTrue(
+      message =
+          "Required parameters for SnowflakeToGCS not passed. "
+              + "Template property cannot be provided for BOTH the input database, schema, table AND an equivalent query. "
+              + "Refer to snowflake/README.md for more instructions.")
+  private boolean isPropertiesBothSet() {
+    return (StringUtils.isBlank(sfQuery)
+        || (StringUtils.isBlank(sfDatabase)
+            && StringUtils.isBlank(sfSchema)
+            && StringUtils.isBlank(sfTable)));
```
Comment (vanshaj-bhatia) [inline java/src/main/java/com/google/cloud/dataproc/templates/snowflake/SnowflakeToGCSConfig.java:164]: This condition doesn't seem correct. Please review.

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

**Review by vanshaj-bhatia — APPROVED** [review]
LGTM

**Pull request merged**